newIDE/app/src/EventsFunctionsList/EventsBasedBehaviorTreeViewItemContent.js

```diff
@@ -352,7 +360,29 @@ export class EventsBasedBehaviorTreeViewItemContent
     this.props.forceUpdate();
   }
 
-  getRightButton() {
-    return null;
+  getRightButton(i18n: I18nType) {
+    return {
+      icon: <Add />,
+      label: i18n._(t`Add a function`),
+      click: () => this.addFunctionAtSelection(),
+    };
+  }
+
+  addFunctionAtSelection(): void {
+    const { selectedEventsFunction, selectedEventsBasedBehavior } = this.props;
+    const eventsFunctionsContainer = this.eventsBasedBehavior.getEventsFunctions();
+    const index =
+      selectedEventsBasedBehavior === this.eventsBasedBehavior &&
```

Comment:
Can you explain in which case the selectedEventsBasedBehavior would not be the this.eventsBasedBehavior?
I'm a bit confused when this could be the case.

Reply:
If you selected a function in one behavior but add a function in another behavior.
I'll add a comment.